coursera/cookies.py

```diff
@@ -108,6 +115,39 @@ def prepare_auth_headers(session, include_cauth=False):
     return headers
 
 
+def login_with_webview():
+    """
+    Creates a physical login window for the use to login.
+    """
+    url = "https://www.coursera.org/?authMode=login"
+
+    def get_cookies():
+        timeout = 120
+
+        def is_logged_in():
+            return webview.evaluate_js("document.querySelector('#logout-btn') !== null")
+
+        while not webview.webview_ready() or not is_logged_in():
+            time.sleep(1)
+            timeout -= 1
+            if timeout == 0:
+                raise AuthenticationFailed("Can't login, timeout exceeded.")
+
+        cookies = webview.evaluate_js("document.cookie")
+        webview.destroy_window()
+        return cookies
+
+    def create_login_window():
+        webview.create_window("Login to Coursera", url)
+
+    pool = ThreadPool(processes=1)
+    async_result = pool.apply_async(get_cookies)
+    create_login_window()
+    cookies = async_result.get()
+    parsed_cookies = dict(re.findall(r"([^\=]+)\=([^;]+);\s?", cookies))
```

Comment:
Could you please add a comment and describe how the values you're looking for look like? It's a bit difficult to decipher from the regexp.

Reply:
> Given how annoying it is now to login and enter Captcha, maybe it makes sense to save cookies by default so that the user does not have to login on every run?

In my case it will open the window, but closes it immediately. I did think of caching the result somewhere in our own .coursera-dl config file, so that the regular login function can try to login with those and otherwise try it again with the webview.